File: 주은/W2/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    int x, y;
    
    for (int i = 1; i <= brown/2; i++) {
        x = i;
        y = brown/2-i;
        if (y-2 <= 0) continue;
        if (x*(y-2) == yellow) {
            answer.push_back(y);
            answer.push_back(x+2);
            break;
        }
    }
    
    return answer;
}

File: 채현/W2/카펫.py
def solution(brown, yellow):
    n = brown + yellow
    
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            h = i
            w = n // i

            if (w - 2) * (h - 2) == yellow:
                return [w, h]
    

# brown + yellow 는 전체 칸수 (어차피 가로 세로는 전체 칸수의 약수쌍밖에 안됨)
# yellow는 (w - 2) * (h - 2)
